anta/models.py

```diff
@@ -448,15 +448,16 @@ def __init__(
         self.device: AntaDevice = device
         self.inputs: AntaTest.Input
         self.instance_commands: list[AntaCommand] = []
-        self.result: TestResult = TestResult(
-            name=device.name,
-            test=self.name,
-            categories=self.categories,
-            description=self.description,
-        )
+        self.result: TestResult = TestResult(name=device.name, test=self.name, categories=self.categories, description=self.description)
         self._init_inputs(inputs)
-        if self.result.result == AntaTestStatus.UNSET:
+        if hasattr(self, "inputs"):
```

Comment:
this looks breaking - if there is no `inputs` then we don't init the command anymore?

Reply:
This is not breaking.
The first condition was to verify that `self.result.result` has its default value (UNSET) and has not been set to ERROR because of input validation error.
I find the condition `hasattr(self, "inputs")` more readable and completely equivalent.